src/app/store/actions/VehicleTechRecordModel.actions.spec.ts

```diff
@@ -1,81 +1,135 @@
 import {
-  EVehicleTechRecordModelActions,
-  GetVehicleTechRecordModel,
-  GetVehicleTechRecordModelSuccess,
-  GetVehicleTechRecordModelHavingStatusAll,
-  GetVehicleTechRecordModelHavingStatusAllSuccess,
-  SetVehicleTechRecordModelOnCreate
+  EVehicleTechRecordActions,
+  GetVehicleTechRecordHavingStatusAll,
+  GetVehicleTechRecordHavingStatusAllSuccess,
+  SetVehicleTechRecordOnCreate,
+  UpdateVehicleTechRecord,
+  UpdateVehicleTechRecordSuccess,
+  SetSelectedVehicleTechnicalRecord,
+  SetViewState
 } from './VehicleTechRecordModel.actions';
 import {
   VehicleIdentifiers,
-  VehicleTechRecordModel
+  VehicleTechRecordModel,
+  VehicleTechRecordEdit,
+  VehicleTechRecordState
 } from '@app/models/vehicle-tech-record.model';
 import { TechRecord } from '@app/models/tech-record.model';
 import { SearchParams } from '@app/models/search-params';
+import { RECORD_STATUS, VIEW_STATE } from '@app/app.enums';
 
-const vehicleTechRecordModel: VehicleTechRecordModel = {} as VehicleTechRecordModel;
-const techRecord: TechRecord = {} as TechRecord;
+const mockTechRecord: TechRecord = {
+  statusCode: RECORD_STATUS.CURRENT
+} as TechRecord;
 
-describe('GetVehicleTechRecordModel', () => {
-  it('the action should have the right type and payload', () => {
-    const action = new GetVehicleTechRecordModel(techRecord);
+const mockVehicleTechRecord: VehicleTechRecordModel = {
+  techRecord: [mockTechRecord]
+} as VehicleTechRecordModel;
+
+const mockVehicleRecordEdit: VehicleTechRecordEdit = {
+  primaryVrm: 'vrm1',
+  techRecord: [mockTechRecord]
+} as VehicleTechRecordEdit;
+
+describe('GetVehicleTechRecordHavingStatusAll', () => {
+  it('should create GetVehicleTechRecordHavingStatusAll action', () => {
```

Comment:
We could be adding with the correct payload as this is what this action creator does so it exactly describes what this is about.
it takes a payload in that will then be consumed in the reducer.

Reply:
@JonLazariniDD  what action does mainly is to create `an instance` and the output should be inferred in the assertion i.e. creating the correct type with or without payload. Specifying the creation of the `correct payload in the title` does not necessarily convey much meaning as the encapsulated info is of less importance at this stage but just the type which again the tests documents and asserts

Reply:
What I was trying to point out is that it is not explicitly stating what the redux (elm) architecture is about.
This particular action creator takes a payload (searchParams) as an argument and returns a new object containing this payload and its relevant action type bound to it. We are not capturing the 'moving part' of this action creator which is the payload I think. Also, probably down to personal preference but I am not a fan of having the payload property in the action as payload as it is not explicit I find, searchParams has more meaning than payload if we had to follow this action through the codebase. It is the convention we have for now and can stick with it.
So up to you how you wish to update or not but here we could be assuming that developers would have a knowledge of the flux/redux architecture and if I only was to read the description of the this particular test I wouldn't know what it does without looking at the implementation.
This could be overthought from me but that was my intent behind this comment.